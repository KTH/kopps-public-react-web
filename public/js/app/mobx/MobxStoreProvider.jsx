/* eslint no-use-before-define: ["error", "nofunc"] */

import React from 'react'

import { createMobxManager } from './manager'

export default MobxStoreProvider

/**
 * React-component which prepares a MobX-store
 * so that its data can be used with the useStore() hook
 * of any other component embedded into this Provider.
 *
 * @param {object} props
 * @param {function} props.initCallback
 * @param {string} [props.storeId]
 * @param {object} props.children
 *
 */
function MobxStoreProvider(props) {
  const { initCallback, storeId, children } = props

  const manager = createMobxManager(storeId || 'default', initCallback)

  const { Provider: ReactContextProvider, value: mobxStore } = manager.getProviderComponentAndValue()

  return <ReactContextProvider value={mobxStore}>{children}</ReactContextProvider>
}
