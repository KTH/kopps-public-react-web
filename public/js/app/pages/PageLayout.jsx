import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
