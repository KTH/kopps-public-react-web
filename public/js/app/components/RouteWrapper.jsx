/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'

function RouteWrapper({
  createBreadcrumbs,
  component: Component,
  layout: Layout,
  applicationStore,
  createMenuData,
  ...rest
}) {
  const menuData = createMenuData(applicationStore)
  const breadcrumbs = createBreadcrumbs(applicationStore)
  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Route
        {...rest}
        render={() => (
          <Layout breadcrumbs={breadcrumbs} menuData={menuData}>
            <Component />
          </Layout>
        )}
      />
    </MobxStoreProvider>
  )
}

RouteWrapper.propTypes = {
  createBreadcrumbs: PropTypes.func,
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.object.isRequired,
}

RouteWrapper.defaultProps = {
  createBreadcrumbs: () => ({ include: 'student', items: [] }),
}

export default RouteWrapper
