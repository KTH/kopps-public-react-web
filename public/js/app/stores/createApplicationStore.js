/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
