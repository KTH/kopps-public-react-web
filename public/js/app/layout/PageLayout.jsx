import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
