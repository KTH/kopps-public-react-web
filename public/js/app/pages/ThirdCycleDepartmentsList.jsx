import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'reactstrap'
import { PageHeading, Heading, LinkList, Link, Text } from '@kth/kth-reactstrap/dist/components/studinfo'
import { CollapseDetails } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import Article from '../components/Article'
import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import translate from '../../../../domain/translate'
import { localeCompareDepartments } from '../../../../domain/departments'
import { replaceSiteLinkForThirdCyclePages, thirdCycleDepartmentLink } from '../util/links'

function DepartmentsLinkListResearch({ departments }) {
  const { language } = useStore()
  departments.sort(localeCompareDepartments(language))
  return (
    <LinkList>
      {departments.map(department => (
        <Link key={department.name} href={thirdCycleDepartmentLink(department.code, language)}>
          {department.name}
        </Link>
      ))}
    </LinkList>
  )
}

function CurrentSchools() {
  const { currentSchoolsWithDepartments } = useStore()

  return (
    <>
      {currentSchoolsWithDepartments.map(school => (
        <Fragment key={school.name}>
          <Heading size="h2" text={school.name} />
          <DepartmentsLinkListResearch departments={school.departments} />
        </Fragment>
      ))}
    </>
  )
}

function DeprecatedSchools() {
  const { language, deprecatedSchoolsWithDepartments } = useStore()
  const t = translate(language)
  return (
    <>
      <h2>{t('departments_deprecated_schools')}</h2>
      <CollapseDetails title={t('departments_deprecated_schools_collapsible')}>
        {deprecatedSchoolsWithDepartments.map(school => (
          <Fragment key={school.name}>
            <Heading size="h3" text={school.name} />
            <DepartmentsLinkListResearch departments={school.departments} />
          </Fragment>
        ))}
      </CollapseDetails>
    </>
  )
}

function ThirdCycleDepartmentsList() {
  const { language } = useStore()
  const t = translate(language)
  const pageName = t('third_cycle_courses_by_school')
  const pageDescription = t('third_cycle_courses_by_school_description')

  React.useEffect(() => {
    let isMounted = true
    if (isMounted) replaceSiteLinkForThirdCyclePages(pageName, language)
    return () => (isMounted = false)
  })

  return (
    <>
      <Row>
        <Col>
          <PageHeading>{pageName}</PageHeading>
        </Col>
      </Row>
      <Row>
        <Col>
          <p>{pageDescription}</p>
        </Col>
      </Row>
      <Row>
        <Col>
          <Article classNames={['paragraphs']}>
            <CurrentSchools />
            <DeprecatedSchools />
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

DepartmentsLinkListResearch.propTypes = {
  departments: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      code: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
}

export default ThirdCycleDepartmentsList
