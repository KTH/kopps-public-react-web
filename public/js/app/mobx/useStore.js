/* eslint no-use-before-define: ["error", "nofunc"] */

import createApplicationStore from '../stores/createApplicationStore'
import { getMobxManager } from './manager'

export default useStore

/**
 * Hook which can be used by any React component
 * within the related MobxStoreProvider.
 *
 * @param {string} [storeId]
 *
 *      Current state of application store
 */
function useStore(storeId = 'default') {
  const manager = getMobxManager(storeId)

  const isTestEnvironmentWithoutProviderComponent = manager == null
  if (isTestEnvironmentWithoutProviderComponent) {
    const dummyStoreForTesting = createApplicationStore()
    return dummyStoreForTesting
  }

  return manager.getContextHook()
}
