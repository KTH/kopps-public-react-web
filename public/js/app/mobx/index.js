/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
