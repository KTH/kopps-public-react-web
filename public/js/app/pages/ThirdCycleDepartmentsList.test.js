/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import { StaticRouter } from 'react-router'
import { MobxStoreProvider } from '../mobx'

import ThirdCycleDepartmentsList from './ThirdCycleDepartmentsList'
import RouteWrapper from '../components/RouteWrapper'
import PageLayout from '../layout/PageLayout'
import getThirdCycleMenuData from '../config/thirdCycleMenuData'
import getThirdCycleBreadcrumbs from '../config/thirdCycleBreadcrumbs'

import createApplicationStore from '../stores/createApplicationStore'

import commonSettings from '../config/mocks/mockCommonSettings'

expect.extend(toHaveNoViolations)

const applicationStore = createApplicationStore()

const { proxyPrefixPath } = commonSettings
const testCurrentSchoolsWithDepartments = {
  sv: [
    {
      departmentPrefix: 'A',
      name: 'Skola',
      departments: [{ code: 'BB', name: 'Avdelning' }],
    },
  ],
  en: [
    {
      departmentPrefix: 'A',
      name: 'School',
      departments: [{ code: 'BB', name: 'Department' }],
    },
  ],
}

const WrapperThirdCycleDepartmentsList = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })
  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <RouteWrapper
        exact
        path="/utbildning/forskarutbildning/kurser/avdelning"
        component={ThirdCycleDepartmentsList}
        layout={PageLayout}
        applicationStore={updatedApplicationStore}
        createBreadcrumbs={store => ({
          include: 'university',
          items: getThirdCycleBreadcrumbs(store),
        })}
        createMenuData={store => ({
          selectedId: 'thirdCycleDepartmentsList',
          ...getThirdCycleMenuData(store),
        })}
      />
    </StaticRouter>
  )
}

const ThirdCycleDepartmentsListWithLayout = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })
  applicationStore.setCurrentSchoolsWithDepartments(testCurrentSchoolsWithDepartments[lang])
  const menuData = getThirdCycleMenuData(applicationStore)
  const breadcrumbs = getThirdCycleBreadcrumbs(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout
          breadcrumbs={{
            include: 'university',
            items: breadcrumbs,
          }}
          menuData={{ selectedId: 'thirdCycleDepartmentsList', ...menuData }}
        >
          <ThirdCycleDepartmentsList />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component ThirdCycleDepartmentsList within RouterWrapper', () => {
  test('simple render', done => {
    render(<WrapperThirdCycleDepartmentsList lang="en" />)
    done()
  })
})

describe('Render component ThirdCycleDepartmentsList within Layout', () => {
  test('verify that page is accessible', async () => {
    const { container } = render(<ThirdCycleDepartmentsListWithLayout lang="en" />)
    expect(await axe(container)).toHaveNoViolations()
  })

  test('get page header in English', () => {
    render(<ThirdCycleDepartmentsListWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Third-cycle courses by school')
  })

  test('get page header in Swedish', () => {
    render(<ThirdCycleDepartmentsListWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Forskarkurser per skola')
  })

  test('match to snapshot in English', done => {
    const { asFragment } = render(<ThirdCycleDepartmentsListWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<ThirdCycleDepartmentsListWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})

describe('Render component ThirdCycleDepartmentsList and check its menu, content and links', () => {
  test('check all links on the page in English', () => {
    render(<ThirdCycleDepartmentsListWithLayout lang="en" />)

    const h2Header = screen.getByRole('heading', { level: 2 })
    expect(h2Header).toHaveTextContent('School')

    const links = screen.getAllByRole('link')
    expect(links.length).toBe(6)
    // Menu links
    expect(links[0]).toHaveTextContent('PhD studies')
    expect(links[0].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/?l=en')
    expect(links[1]).toHaveTextContent('Third-cycle courses by school')
    expect(links[1].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/avdelning?l=en')
    expect(links[2]).toHaveTextContent('Search third-cycle courses')
    expect(links[2].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/sok?l=en')
    // Main content links
    expect(links[3]).toHaveTextContent('Department')
    expect(links[3].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/org/BB?l=en')
    // Footer links
    expect(links[4]).toHaveTextContent('Central study counseling')
    expect(links[4].href).toStrictEqual('https://www.kth.se/en/studies/master/general-study-counselling-1.621634')
    expect(links[5]).toHaveTextContent('kopps@kth.se')
    expect(links[5].href).toStrictEqual('mailto:kopps@kth.se')
  })

  test('check all links on the page in Swedish', () => {
    render(<ThirdCycleDepartmentsListWithLayout lang="sv" />)

    const h2Header = screen.getByRole('heading', { level: 2 })
    expect(h2Header).toHaveTextContent('Skola')

    const links = screen.getAllByRole('link')
    expect(links.length).toBe(6)
    // Menu links
    expect(links[0]).toHaveTextContent('Forskarutbildning')
    expect(links[0].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/')
    expect(links[1]).toHaveTextContent('Forskarkurser per skola')
    expect(links[1].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/avdelning')
    expect(links[2]).toHaveTextContent('Sök forskarkurs')
    expect(links[2].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/sok')
    // Main content links
    expect(links[3]).toHaveTextContent('Avdelning')
    expect(links[3].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser/org/BB')
    // Footer links
    expect(links[4]).toHaveTextContent('Central studievägledning')
    expect(links[4].href).toStrictEqual('https://www.kth.se/utbildning/traffakth/studievagledning')
    expect(links[5]).toHaveTextContent('kopps@kth.se')
    expect(links[5].href).toStrictEqual('mailto:kopps@kth.se')
  })
})
