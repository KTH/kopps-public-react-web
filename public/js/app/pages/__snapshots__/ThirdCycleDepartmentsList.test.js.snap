// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component ThirdCycleDepartmentsList within Layout match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-label="Sub menu"
      class="col navbar navbar-expand-lg navbar-light"
      id="mainMenu"
    >
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="nav"
        >
          <li
            class="parentLink"
          >
            <a
              href="/utbildning/forskarutbildning/?l=en"
            >
              PhD studies
            </a>
          </li>
        </ul>
        <ul
          class="nav nav-list expandable"
        >
          <li
            class="nav-item ancestor"
          >
            <div>
              Courses
            </div>
          </li>
          <li
            class="nav-item leaf selected"
          >
            <a
              class="nav-link"
              href="/utbildning/forskarutbildning/kurser/avdelning?l=en"
            >
              Third-cycle courses by school
            </a>
          </li>
          <li
            class="nav-item leaf"
          >
            <a
              class="nav-link"
              href="/utbildning/forskarutbildning/kurser/sok?l=en"
            >
              Search third-cycle courses
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="col"
    >
      <main
        id="mainContent"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <header
              role="presentation"
            >
              <h1>
                Third-cycle courses by school
              </h1>
            </header>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <article
              class="article paragraphs"
            >
              <h2>
                School
              </h2>
              <ul
                class="link-list"
              >
                <li>
                  <a
                    href="/utbildning/forskarutbildning/kurser/org/BB?l=en"
                  >
                    Department
                  </a>
                </li>
              </ul>
              <h2>
                Older Schools
              </h2>
              <details
                class=""
              >
                <summary
                  aria-label=""
                  class="white"
                >
                  Courses older schools
                </summary>
                <div />
              </details>
            </article>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <footer
              class="border-top mt-4 pt-1"
              id="articleFooter"
            >
              <address>
                Contact for questions regarding education at KTH: 
                <a
                  href="https://www.kth.se/en/studies/master/general-study-counselling-1.621634"
                >
                  Central study counseling
                </a>
                <br />
                Contact regarding technical matters on this page: 
                <a
                  href="mailto:kopps@kth.se"
                >
                  kopps@kth.se
                </a>
              </address>
            </footer>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render component ThirdCycleDepartmentsList within Layout match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-label="Undermeny"
      class="col navbar navbar-expand-lg navbar-light"
      id="mainMenu"
    >
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="nav"
        >
          <li
            class="parentLink"
          >
            <a
              href="/utbildning/forskarutbildning/"
            >
              Forskarutbildning
            </a>
          </li>
        </ul>
        <ul
          class="nav nav-list expandable"
        >
          <li
            class="nav-item ancestor"
          >
            <div>
              Kurser
            </div>
          </li>
          <li
            class="nav-item leaf selected"
          >
            <a
              class="nav-link"
              href="/utbildning/forskarutbildning/kurser/avdelning"
            >
              Forskarkurser per skola
            </a>
          </li>
          <li
            class="nav-item leaf"
          >
            <a
              class="nav-link"
              href="/utbildning/forskarutbildning/kurser/sok"
            >
              Sök forskarkurs
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="col"
    >
      <main
        id="mainContent"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <header
              role="presentation"
            >
              <h1>
                Forskarkurser per skola
              </h1>
            </header>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <article
              class="article paragraphs"
            >
              <h2>
                Skola
              </h2>
              <ul
                class="link-list"
              >
                <li>
                  <a
                    href="/utbildning/forskarutbildning/kurser/org/BB"
                  >
                    Avdelning
                  </a>
                </li>
              </ul>
              <h2>
                Äldre skolor
              </h2>
              <details
                class=""
              >
                <summary
                  aria-label=""
                  class="white"
                >
                  Kurser äldre skolor
                </summary>
                <div />
              </details>
            </article>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <footer
              class="border-top mt-4 pt-1"
              id="articleFooter"
            >
              <address>
                Kontakt för frågor om utbildning på KTH: 
                <a
                  href="https://www.kth.se/utbildning/traffakth/studievagledning"
                >
                  Central studievägledning
                </a>
                <br />
                Teknisk kontakt om denna sida: 
                <a
                  href="mailto:kopps@kth.se"
                >
                  kopps@kth.se
                </a>
              </address>
            </footer>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
